### PR TITLE
JAMES-1605 fix AbstractRecipientRewriteTable.mapAddress to respect or…

### PR DESCRIPTION
**Pull request by mbaechler**
…dering

### CODE REVIEW

**Comment by rouazana** [comment]
:+1:

File: data/data-library/src/test/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTableTest.java [hunk 1]
```diff
@@ -294,6 +300,40 @@ public void testAliasDomainMapping() throws ErrorMappingException, RecipientRewr
...
+        String singleDomainAlias = RecipientRewriteTable.ALIASDOMAIN_PREFIX + "first";
+        assertEquals(singleDomainAlias, AbstractRecipientRewriteTable.sortMappings(singleDomainAlias));
+    }
+     
+    @Test
+    public void sortMappingsShouldReturnSameStringWhenTwoDomainAliases() {
+        String firstAliasMapping = RecipientRewriteTable.ALIASDOMAIN_PREFIX + "first";
+        String secondAliasMapping = RecipientRewriteTable.ALIASDOMAIN_PREFIX + "second";
+        String mappings = RecipientRewriteTableUtil.CollectionToMapping(Arrays.asList(firstAliasMapping, secondAliasMapping));
+        assertEquals(mappings, AbstractRecipientRewriteTable.sortMappings(mappings));
```
Comment (aduprat) [inline data/data-library/src/test/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTableTest.java]: It's not clear what you are expecting; but it's ok as you are not testing _CollectionToMapping_.

**Comment by aduprat** [comment]
:+1:

**Comment by chibenwa** [comment]
+1

**Comment by mbaechler** [comment]
already applied

**Pull request closed without merging**